osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/SliderSelectionBlueprint.cs

```diff
@@ -11,22 +13,45 @@ namespace osu.Game.Rulesets.Osu.Edit.Blueprints.Sliders
 {
     public class SliderSelectionBlueprint : OsuSelectionBlueprint
     {
-        private readonly SliderCircleSelectionBlueprint headBlueprint;
+        protected SliderCircleSelectionBlueprint HeadBlueprint { get; private set; }
 
         public SliderSelectionBlueprint(DrawableSlider slider)
             : base(slider)
         {
+            Container sliderCircleContainer;
             var sliderObject = (Slider)slider.HitObject;
 
             InternalChildren = new Drawable[]
             {
                 new SliderBodyPiece(sliderObject),
-                headBlueprint = new SliderCircleSelectionBlueprint(slider.HeadCircle, sliderObject, SliderPosition.Start),
-                new SliderCircleSelectionBlueprint(slider.TailCircle, sliderObject, SliderPosition.End),
+                sliderCircleContainer = new Container
+                {
+                    Children = new[]
+                    {
+                        HeadBlueprint = new SliderCircleSelectionBlueprint(slider.HeadCircle, sliderObject, SliderPosition.Start),
+                        new SliderCircleSelectionBlueprint(slider.TailCircle, sliderObject, SliderPosition.End),
+                    }
+                },
                 new PathControlPointVisualiser(sliderObject),
             };
+
+            ScheduledDelegate regenerationDebounce = null;
+
+            sliderObject.OnRegenerated += () =>
+            {
+                regenerationDebounce?.Cancel();
```

Comment:
why scheduled? why debounced?

Reply:
This one's a bit weird, but I don't know a better way of doing it - the blueprint's debounced schedule happens in `Update()`, which happens towards the end of the frame. Meaning if nested children are changed, their alive states aren't recomputed until the next frame, resulting in the ticks flashing for one frame.

Open to suggestions (apart from comments being required ofc).

Reply:
If ApplyDefaults happens solely from inside the slider, via some `PathChanged` event, then it might work out fine, although DrawableSlider will still have a schedule for debounce.

Reply:
Why is there no comment explaining this? Either way, we can't have all these random debounce things on non-asynchronous operations. that's not what debouncing should be used for. a schedule alone with a comment is enough, but adding debounce suggests logical flaw.